### ISSUE TITLE
Editing a pipeline crushes CellProfiler

### ISSUE THREAD

**Issue by vkhodygo** [CellProfiler/CellProfiler#4443]
**Describe the bug**
CellProfiler crushes when I try to switch positions of modules in a pipeline via drag-and-drop.

**To Reproduce**
1. Start CellProfiler
2. Use any (non-default) pipeline
3. Try to change the order of modules

**Expected behavior**
No problems should be observed, this is a common functionality.

**Desktop (please complete the following information):**
 - OS: ArchLinux, Python 3.9 
 - CellProfiler Version 4.2.1 (latest git commit)
 - source
 - cloned

**Additional context**
Error text:
```
#
# A fatal error has been detected by the Java Runtime Environment:
#
#  SIGSEGV (0xb) at pc=0x00007f4f161d7c27, pid=29767, tid=29767
#
# JRE version: OpenJDK Runtime Environment (11.0.12+7) (build 11.0.12+7)
# Java VM: OpenJDK 64-Bit Server VM (11.0.12+7, mixed mode, tiered, compressed oops, g1 gc, linux-amd64)
# Problematic frame:
# C  [libpython3.9.so.1.0+0x10fc27]  _PyObject_GC_New+0x27
#
# Core dump will be written. Default location: Core dumps may be processed with "/usr/lib/systemd/systemd-coredump %P %u %g %s %t %c %h" (or dumping to /tmp/tmpojp0pmdg/core.29767)
#
# An error report file with more information is saved as:
# /tmp/tmpojp0pmdg/hs_err_pid29767.log
#
# If you would like to submit a bug report, please visit:
#   https://bugreport.java.com/bugreport/crash.jsp
# The crash happened outside the Java Virtual Machine in native code.
# See problematic frame for where to report the bug.
#

zsh: killed     cellprofiler -L DEBUG

```
**Attached files**
[hs_err_pid29767.log](https://github.com/CellProfiler/CellProfiler/files/6931093/hs_err_pid29767.log)

**Comment by DavidStirling** [comment]
Hi @vkhodygo,

This is an issue we've encountered with wxPython 4.1.1, would you happen to be using that? It should be pinned to 4.1.0. Otherwise I'd recommend trying Python 3.8, since there are currently some compatability issues with 3.9.

**Comment by vkhodygo** [comment]
Hi @DavidStirling,

Colour me surprised but no, I use `4.1.0` installed via `pip`. 
Could it be related to `wxgtk` which is a dependency of `wxPython`? 

Yes, I know that you recommend 3.8 but I've run some tests and it seems to work fine (some 3.8 related issues disappeared as well). Dealing with virtual environments on top of other things is too much at the moment, unfortunately.

**Comment by DavidStirling** [comment]
Do you mean that you still had the issue on 3.8 + 4.1.0?

I understand that it's a pain to handle virtual environments. Something in recent versions of wx broke the functionality for building custom drag/drop objects, which is critical for manipulating the pipeline. The lack of updates to wx recently has been the primary issue preventing full support for Python 3.9.

**Comment by vkhodygo** [comment]
No, I did run some tests on my Ubuntu VM and everything seems to be fine, that's a shame.

> critical for manipulating the pipeline

May be it's just easier to add buttons that move elements of the pipeline up and down as a workaround?

P.S. I did check the wxPython repository and indeed, this is a common issue. However. it looks like there is a new release coming as there is a branch with `4.1.2` updates.

**Comment by DavidStirling** [comment]
> May be it's just easier to add buttons that move elements of the pipeline up and down as a workaround?

These buttons already exist, just below the pipeline list.

![image](https://user-images.githubusercontent.com/26802537/128214846-4b737887-dff2-402b-8eb3-2bf95fa8e6fc.png)

I think some users on Linux have trouble with the text icons not showing, but they should still work and move the selected modules.

While this functions as a workaround, we felt that we can't really make a release where dragging a module causes the entire program to crash.

> I did check the wxPython repository and indeed, this is a common issue. However. it looks like there is a new release coming as there is a branch with 4.1.2 updates.

This is fantastic news, thanks for pointing that out!

**Comment by vkhodygo** [comment]
Welp, that resolved my other issue: as you mentioned the icons are missing (on both my machines) so I didn't check the actual functionality of the buttons. 

> While this functions as a workaround, we felt that we can't really make a release where dragging a module causes the entire program to crash.

Fair enough, that means we should patiently wait.

**Comment by gnodar01** [comment]
Closing this as the drag-and-drop issue no longer exists in CP5 (currently python 3.9 and wxpython 4.2.2), and the blank button is captured by #4986.
